src/lib/core/overlay/overlay.scss

```diff
@@ -20,7 +20,6 @@
     left: 0;
     height: 100%;
     width: 100%;
-    z-index: $md-z-index-overlay-container;
```

Comment:
I think it HAS to be removed, otherwise any inner z-index will be ignored.

See http://jsbin.com/mesujat/1/edit?html,css,output yellow is over green, despite it having thousand times higher z-index.